src/rocks_compaction_operation.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *    Copyright (C) 2017 Percona, LLC.
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    As a special exception, the copyright holders give permission to link the
+ *    code of portions of this program with the OpenSSL library under certain
+ *    conditions as described in each individual source file and distribute
+ *    linked combinations including the program with the OpenSSL library. You
+ *    must comply with the GNU Affero General Public License in all respects for
+ *    all of the code used other than as permitted herein. If you modify file(s)
+ *    with this exception, you may extend this exception to your version of the
+ *    file(s), but you are not obligated to do so. If you do not wish to do so,
+ *    delete this exception statement from your version. If you delete this
+ *    exception statement from all source files in the program, then also delete
+ *    it in the license file.
+ */
+
+#pragma once
+
+#include <rocksdb/db.h>
+#include <rocksdb/slice.h>
+
+namespace mongo {
+
+    class RocksCompactionOperation {
```

Comment:
Let's start with:
1. Simple struct storing all the fields by value and serving as DTO, like:
```
struct RocksCompacitonOperation {
std::string _start_slice;
std::string _end_slice;

bool rangeDropped;
};
```
2. Consider having regular & move constructors for the struct.
3. Put this struct into `RocksCompactionScheduler` file (maybe even as a subclass in the scheduler).
4. The background compaction thread should determine all the needed info from the struct and execute operations in a procedural style.

When we need something complex - we may think of switching to this sort of class with own operational logic, but before that - YAGNI principle is a rule.